TSPL.docc/ReferenceManual/Attributes.md

```diff
@@ -1304,6 +1304,53 @@ can increase your binary size and adversely affect performance.
   because of the larger symbol table slowing dyld down.
 -->
 
+### preconcurrency
+
+Apply this attribute to a declaration,
+to suppress strict concurrency checking.
+
+On an import declaration,
+this attribute reduces the strictness of concurrency checking
+for code that uses types from the imported module.
+Specifically,
+types from the imported module
+that aren't explicitly marked as either sendable or nonsendable
+can be used in a context that requires sendable types.
+
+On other declarations,
+this attribute reduces the strictness of concurrency checking
+for code that uses the symbol being declared.
+When you use this symbol in a scope that has minimal concurrency checking,
+you can use members of this type
+that aren't explicitly marked as either sendable or nonsendable
+in a context that requires sendable types.
+You can apply this attribute
+to the following kinds of declarations:
+
+- Structures, classes, and actors
+- Enumerations and enumeration cases
+- Protocols
+- Variables and constants
+- Subscripts
+- Initializers
+- Functions
+
+You can use this attribute as follows,
+to aid in migrating code to strict concurrency checking:
+
+1. Enable strict checking.
+1. Annotate imports and declarations that need to be migrated
+   with the `preconcurrency` attribute.
+1. After migrating a declaration,
+   remove the `preconcurrency` attribute from it.
```

Comment:
`@preconcurrency` on declarations has a different role than on imports. The flow you describe is the right one for imports, although I'd collapse the middle two steps:
1. Enable strict checking
2. Annotate imports with `@preconcurrency` for imports of libraries that haven't enabled strict checking
3. Once an imported library has enabled strict checking, remove `@preconcurrency` from the import. (The compiler will also produce a warning)

For a declaration, add `@preconcurrency` when you are adding concurrency-related constraints to the declaration but you still have clients that haven't migrated to Swift concurrency. Remove the `@preconcurrency` when all of your clients have migrated.